typescript/sdk/src/token/EvmWarpCrudModule.ts

```diff
@@ -0,0 +1,216 @@
+  ): Promise<EthersV5Transaction[]> {
+    return [
+      ...(await this.updateIsm(config)),
+      ...(await this.updateHook(config)),
+    ];
+  }
+
+  /**
+   * Updates an existing Warp route ISM with a given configuration.
+   *
+   * This method handles two cases:
+   * 1. If the `config.interchainSecurityModule` is an object,
+   *  - Checks if the current onchain ISM configuration matches the provided configuration.
+   *  - If not, it deploys a new ISM module, and updates the contract's ISM.
+   * 2. If the `config.interchainSecurityModule` is a string
+   *  - It attempts to derive the ISM from the provided string, and updates the contract's ISM.
+   *
+   * @param config - The token router configuration, including the ISM configuration.
+   * @returns An array of Ethereum transactions that need to be executed to update the ISM configuration.
+   */
+  async updateIsm(
+    config: DerivedTokenRouterConfig,
+  ): Promise<EthersV5Transaction[]> {
+    const transactions: EthersV5Transaction[] = [];
+
+    const contractToUpdate = await this.args.addresses[config.type].deployed();
+
+    if (typeof config.interchainSecurityModule === 'string') {
+      // Derive & set ISM
+      const ism = await this.reader.evmIsmReader.deriveIsmConfig(
+        config.interchainSecurityModule,
+      );
+      transactions.push({
+        transaction:
+          await contractToUpdate.populateTransaction.setInterchainSecurityModule(
+            ism.address,
+          ),
+        type: ProviderType.EthersV5,
+      });
+    } else if (typeof config.interchainSecurityModule === 'object') {
+      const onchainConfig = await this.read();
+      if (
+        config.interchainSecurityModule.type !==
+        onchainConfig.interchainSecurityModule!.type
+      ) {
+        // Deploy & set ISM
+        const ismModule = await this.deployIsm(config);
+        transactions.push({
+          transaction:
+            await contractToUpdate.populateTransaction.setInterchainSecurityModule(
+              ismModule,
+            ),
+          type: ProviderType.EthersV5,
+        });
+      }
+    }
+
+    return transactions;
+  }
+
+  /**
+   * Deploys the ISM using the provided configuration.
+   *
+   * @param config - The configuration for the ISM to be deployed.
+   * @returns The deployed ISM contract address.
+   */
+  public async deployIsm(config: DerivedTokenRouterConfig): Promise<string> {
+    // Take the config.ismFactoryAddresses, de-serialize them into Contracts, and pass into EvmIsmModule.create
+    const factories = attachContracts(
+      config.ismFactoryAddresses as HyperlaneAddresses<ProxyFactoryFactories>,
+      proxyFactoryFactories,
+    );
+    const evmIsmModule = await EvmIsmModule.create({
+      chain: this.args.chain,
+      config: config.interchainSecurityModule!,
+      deployer: new HyperlaneProxyFactoryDeployer(this.multiProvider),
+      factories,
+      multiProvider: this.multiProvider,
+    });
```

Comment:
imo the EvmIsmModule should do the attachment of the factories and this should be an "implementation detail" of that module

Reply:
Will have to wait for @paulbalaji to land this change/addition to https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3759